Use fake Org/Task/Epic for guided tours.

jgerigmeyer opened the pull request:
Based on #1467.

![](https://source.unsplash.com/featured/?cute,animal&210924)

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

- Try out all three walkthroughs, but especially "Play" walkthrough
- Try them on a Project that has Epics/Tasks, but also on a Project with no Epics/Tasks

SondraE reviewed (changes requested):
- The first step in the Play walkthrough is partially hidden behind the gray self-guided-tour-is-active bar. 
<img width="435" alt="Screen Shot 2021-09-24 at 11 19 01 AM" src="https://user-images.githubusercontent.com/8933115/134716168-2c84a150-1771-4090-8f02-d0ddffd7ea33.png">

- Also Stacy and I reviewed this PR together, and we made updates to the text for the "Contribute Work" card in the Play walkthrough. We added instructions for who to talk to if the person doesn't have 'push' permissions.

stacyk commented on src/js/components/tour/play.tsx:
```diff
@@ -46,6 +100,34 @@ const PlayTour = (props: TourProps) => {
+      target: '.tour-scratch-org-contribute',
+      title: i18n.t('Contribute your work'),
+      content: (
+        <Trans i18nKey="walkthroughPlayScratchOrgContribute">
+          To contribute your own work from a Scratch Org you need “push”
+          permissions on the Project in GitHub.
+          If you do not have permissions, ask a Project admin.
+          Select “Contribute Work” to
+          create a Task for your Scratch Org. When your Org expires or is
+          deleted, any work not contributed will be lost.
+        </Trans>
```
@SondraE here's 2 of 2 of the text updates we made.

stacyk commented on src/js/components/projects/detail.tsx:
```diff
@@ -435,8 +464,7 @@ const ProjectDetail = (
                       done on this Project and who is working on it. Tasks
-                      represent small changes to the Project, and may be grouped
-                      with other Tasks in an Epic.
+                      represent small changes to the Project, and may be part of an Epic.
```
@SondraE here one of two copy changes we made

jgerigmeyer replied:
@SondraE @stacyk I'm not clear on why this is an improvement. It seems like "may be part of an Epic" is so vague that it's not providing useful information -- why is that better than "may be grouped with other Tasks in an Epic"?

SondraE replied:
@jgerigmeyer We thought "Tasks may be grouped with other Tasks in an Epic" made it sound like users could take a bunch of existing Tasks and pull them together into a group. Probably what should happen here is that we don't try to make these wording changes in this PR, but wait and make them as part of this story: https://trello.com/c/th1Pk6wb/322-i-want-to-better-understand-how-projects-epics-and-tasks-are-related-eg-epics-arent-simply-folders-for-tasks

jgerigmeyer replied:
Makes sense to me. I'm fine with this for now, and we can revisit in the other story. 👍

stacyk reviewed (commented):
Still needs a fix for the popover on the Metecho's `?`  (on the play walkthrough, not self guided) but I have to prep my Mac for repair now.

jgerigmeyer commented:
>     * The first step in the Play walkthrough is partially hidden behind the gray self-guided-tour-is-active bar.

Fixed.

>     * Also Stacy and I reviewed this PR together, and we made updates to the text for the "Contribute Work" card in the Play walkthrough. We added instructions for who to talk to if the person doesn't have 'push' permissions.

👍

dvdherron commented on locales/en/translation.json:
```diff
@@ -495,16 +500,18 @@
-  "walkthroughPlayScratchOrg": "Scratch Orgs are a temporary place for you to view the work on this Project. You can use Scratch Orgs to play with changes to the Project without affecting the Project.",
+  "walkthroughPlayListTasks": "Select the Tasks tab to see a list of all the work being done on this Project and who is doing it. Tasks represent small changes to the Project, and may be part of an Epic. Select a Task and create a Task Scratch Org to view the work on that Task.",
+  "walkthroughPlayMetechoName": "Metecho makes it easier for you to view, test, and contribute to Salesforce Projects without learning GitHub.<1></1><2></2><3>Pronunciation</3>: “Met” rhymes with “bet.” “Echo” as in the reflection of sound waves.<5></5><6>Definition</6>: To share or participate in.",
+  "walkthroughPlayScratchOrg": "This is a sample Scratch Org for this Project. Select “View Org” to see the work on this Project. Your Org will expire after 30 days.",
```
I was tempted to click "View Org" during the walkthrough. Clicking anything in that highlighlighted space closes the walkthrough. Should a click like this do nothing and keep the tour open?

SondraE replied:
I think we should allow people to click active buttons such as "create an epic" during the walkthrough, if they decide they want to go ahead and take that action. On the sample org card, though, it makes less sense to stop the walkthrough when someone clicks that card since the links don't work. Can we can keep the walkthrough open if someone clicks the sample org card?

jgerigmeyer replied:
@SondraE Hm, I have a different leaning here. It's a well-established pattern to disable action clicks during a tour/walkthrough, and as a user I think I'd be surprised if I could take actions while the tour is running. I also don't like the idea of treating some actions differently -- I think they should all behave the same. But we have three options:

1. Continue with the default behavior, where clicks are disabled, but stop the tour.
2. Continue disabling clicks, but don't stop the tour if the user clicks somewhere. To stop the tour the user would need to go through all the steps, or click the "x" to close, or type "Esc".
3. Allow all clicks to function normally, but *remove the grey overlay* while the tour is running. This would require a bit of extra logic to disable clicks on the "fake" org, and isn't my favorite option because I think it'll be more confusing for users.

I would definitely lean toward option 1 or 2.

SondraE replied:
@jgerigmeyer Interesting. I had thought when we were looking at options for building the guided tour, we asked Jason/David whether they wanted the user to be able to click to take the actions at any time during the tour (as demoed in https://reactour.js.org/ which, incidentally, appears to be a little broken right now). I thought that they specifically wanted action clicks to be enabled during the guided tour. That said, I too don't love the idea of treating some actions differently. So given this discussion, I would either pick your option 1 or the option to enable clicks and stop the tour.

jgerigmeyer replied:
I think we can merge this for now, then ask them when we demo this week.

dvdherron reviewed (approved):
LGTM. 

Left a tiny comment about click actions during the tour but otherwise 👍🏽 (Nice touch with the demo tasks/epics @jgerigmeyer )
